corgidrp/spec.py

```diff
@@ -0,0 +1,648 @@
+                            np.abs(yfwhm) / (2 * np.sqrt(2 * np.log(2))) / psf_peakpix_snr)
+
+    return xfit, yfit, gauss2d_xfit, gauss2d_yfit, psf_peakpix_snr, x_precis, y_precis
+
+def compute_psf_centroid(dataset, template_file = None, initial_cent = None, verbose=False, halfwidth=10, halfheight=10):
+    """
+    Compute PSF centroids for a grid of PSFs and return them as a calibration object.
+
+    Args:
+        dataset (Dataset): Dataset containing 2D PSF images. Each image must include pri_hdr and ext_hdr.
+        initial_cent (dict): Dictionary with initial guesses for PSF centroids.
+                             Must include keys 'xcent' and 'ycent', each mapping to an array of shape (N,).
+        template_file (str): filepath of the template PSF, if None, a simulated PSF from the test_data is taken
+        verbose (bool): If True, prints fitted centroid values for each frame.
+        halfwidth (int): Half-width of the PSF fitting box.
+        halfheight (int): Half-height of the PSF fitting box.
+
+    Returns:
+        SpectroscopyCentroidPSF: Calibration object with fitted (x, y) centroids.
+    """
+    if not isinstance(dataset, Dataset):
+        raise TypeError("Input must be a corgidrp.data.Dataset object.")
+    
+    if initial_cent == None:
+        xcent, ycent = None, None
+    else:
+        xcent = np.asarray(initial_cent.get("xcent"))
+        ycent = np.asarray(initial_cent.get("ycent"))
+        if xcent is None or ycent is None:
+            raise ValueError("initial_cent dictionary must contain 'xcent' and 'ycent' arrays.")
+        if len(dataset) != len(xcent) or len(dataset) != len(ycent):
+            raise ValueError("Mismatch between dataset length and centroid guess arrays.")
+    
+    if template_file == None:
+        if 'FILTERS' in dataset[0].pri_hdr:
+            default_file = default_psf_files.get('filtersweep')
+        elif dataset[0].pri_hdr['SLIT'] == 'N':
+            default_file = default_psf_files.get('3d_noslit')
+        else:
+            default_file = default_psf_files.get('3d_slit')
+        test_data_path = os.path.join(os.path.dirname(corgidrp.__path__[0]), "tests", "test_data", "spectroscopy")
+        template_file = os.path.join(test_data_path, 
+            default_file)
+        if verbose:
+            print("no template file given, take simulated file: " + default_file)
+    temp_psf_array = fits.getdata(template_file, ext=0)
+    if len(dataset) != np.shape(temp_psf_array)[0]:
+        raise ValueError("Mismatch between dataset length and template arrays.")
+    try:
+        temp_psf_table = fits.getdata(template_file , ext=1)
+        (xcent_template, ycent_template) = (temp_psf_table['xcent'], temp_psf_table['ycent'])
+        if len(dataset) != len(xcent_template) or len(dataset) != len(ycent_template):
+            raise ValueError("Mismatch between dataset length and template centroid arrays.")
```

Comment:
I don't think this valueerror does anything because it immediately gets silenced by the all-catching except statement in the next line.